src/components/Image/index.js

```diff
@@ -28,11 +30,30 @@ function Image({source: propsSource, isAuthTokenRequired, session, ...forwardedP
         // eslint-disable-next-line react-hooks/exhaustive-deps
     }, [propsSource, isAuthTokenRequired]);
 
+    const imageLoadedSuccessfully = useCallback(
```

Comment:
Hm, I don't think that a name like _"something happened"_ is a conventional way to name a function. Typically it's _"do something"_ or _"on event happened"_ (for event handlers).

Another topic is that we have an internal naming convention **not to** name event handlers _"on something happened"_; Expensify states we should always prefer _"do something"_. I don't agree 100% with this convention, but it's not up for debate.

I believe that an escape path from this situation can be like this:

1. Extract...

```ts
const updateAspectRatio = useCallback(
    (width, height) => {
        if (objectPositionTop) {
            if (width > height) {
                setAspectRatio(1);
                return;
            }
            
            setAspectRatio(height ? width / height : 'auto');
        }
    },
    [objectPositionTop],
);
```

2. Rename `imageLoadedSuccessfully` to `handleLoad` and change the implementation...

```ts
const handleLoad = useCallback(
    (event) => {
        const {width, height} = event.nativeEvent;

        onLoad(event);

        updateAspectRatio(width, height);
    },
    [onLoad],
);
```

This way, everything is named according to what we do, and we stick both to common sense (kind of) and to Expensify naming rules.